dataframes/04-reading-messy-data-into-dataframes.ipynb

```diff
@@ -0,0 +1,502 @@
+{
```

Comment:
<p>I didn't know that <code>dtypes</code>  could be a <code>np.dtype</code>  object. Since not all pandas dtypes have numpy counterparts, this should be a <code>dict</code>  (e.g. <code>{"id": int, "name": str, ...}</code> ) instead IMO.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/dask/dask-examples/pull/157/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Reply:
yeah so I had an issue that it looks like the pd.DataFrame constructor cannot take a compound datatype. So the only other option are to use either:

```python
import io
dtypes={"id": int, "name": str, "x": float, "y": float}
pd.read_csv(io.StringIO(""), names=dtypes.keys(), dtype=dtypes)
```

or 

```python
df = pd.DataFrame({'id':pd.Series([], dtype=str),
                   'name':pd.Series([], dtype=int),
                   'x':pd.Series([], dtype=float),
                   'y':pd.Series([], dtype=float)})
```

Do you prefer either one of these? I think I would go with the first option.

Reply:
<p>Ah, didn't know that. The first one looks best to me.</p>